spring-ai-core/src/main/java/org/springframework/ai/document/Document.java

```diff
@@ -80,31 +90,61 @@ public Document(@JsonProperty("content") String content) {
 		this(content, new HashMap<>());
 	}
 
+	/**
+	 * @deprecated Use builder instead: {@link Document#builder()}.
+	 */
+	@Deprecated(since = "1.0.0-M5", forRemoval = true)
 	public Document(String content, Map<String, Object> metadata) {
```

Comment:
There are probably many users out there using this constructor and perhaps the one with `String id, String content, Map<String, Object> metadata` args.  Despite the having the builder, maybe we keep these ctors?

Reply:
yea, the two are good, once we get to three is when the ambiguity starts so the builder should be preferred